app/assets/javascripts/discourse/app/components/form-template-field/wrapper.js

```diff
@@ -1,17 +1,45 @@
 import Component from "@glimmer/component";
 import Yaml from "js-yaml";
 import { tracked } from "@glimmer/tracking";
+import FormTemplate from "discourse/models/form-template";
+import { action } from "@ember/object";
 
 export default class FormTemplateFieldWrapper extends Component {
   @tracked error = null;
+  @tracked parsedTemplate = null;
 
-  get canShowContent() {
+  constructor() {
+    super(...arguments);
+
+    if (this.args.content) {
+      // Content used when no id exists yet
+      // (i.e. previewing while creating a new template)
+      this._loadTemplate(this.args.content);
+    } else if (this.args.id) {
+      this._fetchTemplate(this.args.id);
+    }
+  }
+
+  _loadTemplate(templateContent) {
     try {
-      const parsedContent = Yaml.load(this.args.content);
-      this.parsedContent = parsedContent;
-      return true;
+      this.parsedTemplate = Yaml.load(templateContent);
     } catch (e) {
       this.error = e;
     }
   }
+
+  @action
+  refreshTemplate() {
+    if (Array.isArray(this.args?.id) && this.args?.id.length === 0) {
```

Comment:
This is a little bit odd. I'm not sure there should be a case where the id is an empty array... (can be addressed in a follow-up).

Reply:
Yeah this one was weird. It was mainly for the case of when there's more than one template in a category and choosing a different template.

It seems when you choose a different template from the dropdown, select-kit removes 1 and then adds 1 so temporarily there is an empty array. So without this check it seems to break things.